include/seccomp.h.in

```diff
@@ -405,6 +405,24 @@ struct seccomp_notif_resp {
 	__s32 error;
 	__u32 flags;
 };
+
+#endif
+
+/* seccomp_notif_addfd and ADDFD_FLAG_SETFD was added in kernel v5.10 */
+#ifndef SECCOMP_ADDFD_FLAG_SETFD
+#define SECCOMP_ADDFD_FLAG_SETFD        (1UL << 0)
+struct seccomp_notif_addfd {
```

Comment:
Why is the `struct seccomp_notif_addfd()` definition inside of the `#ifndef SECCOMP_ADDFD_FLAG_SETFD`?  Seems strange to me.

Reply:
That's a fair question.  The `seccomp_notif_addfd` struct is only used when `SECCOMP_ADDFD_FLAG_SETFD` is present and supported, so using the presence of `SECCOMP_ADDFD_FLAG_SETFD`  to serve as a proxy for the definition of `seccomp_notif_addfd` seems like a reasonable thing to do.

I'm definitely open to other ideas and I'm sure @realsdx would be as well.

https://man7.org/linux/man-pages/man2/seccomp_unotify.2.html